csharp/ql/lib/semmle/code/cil/Types.qll

```diff
@@ -60,11 +60,6 @@ class Class extends ValueOrRefType {
   Class() { this.isClass() }
 }
 
-/** A `record`. */
-class Record extends Class {
-  Record() { this.isRecord() }
-}
-
```

Comment:
Should we delete this? It is probably not used by too many users, but instead of deleting it, we could deprecate it to not introduce a breaking change. Also, we could introduce `RecordClass` instead of `Record`. (We wouldn't need `RecordStruct` because that's indistinguishable from `Struct` in IL.)

Reply:
Alright - thank you! Let me try and re-cap
- We deprecate the <code>Record</code> class in cil/Types.qll and introduce a <code>RecordClass</code> in cil/types.qll (ie. we do not deprecate the support for records in cil, we de-facto rename it instead).
- We deprecate <code>Record</code> in csharp/type.qll and make <code>RecordClass</code> instead (ie. just rename).
Question: Why does it make sense to talk about records (classes) in IL, when they can only be observed by their side effect of having a generated <Clone> method (so basically, I am asking why we shouldn't deprecate the support for "Record" entirely in cil/Types.qll)? It seems that the record concept is native only to C#.

Reply:
> so basically, I am asking why we shouldn't deprecate the support for "Record" entirely in cil/Types.qll?

This is also an option. And probably the more pragmatic option, so let's go this way.